adapters/tlsio_openssl.c

```diff
@@ -897,6 +897,226 @@ static int load_cert_crl_http(
     return rv;
 }
 
+static int atoin(const char *str, int start, int len)
+{
+    int result = 0;
+
+    for (; len > 0; len--, start++)
+    {
+        if (str[start] < '0' || str[start] > '9')
+            return -1;
+
+        result = (result * 10) + (str[start] - '0');
+    }
+
+    return result;
+}
+
+static time_t crl_invalid_after(X509_CRL *crl)
+{
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x20000000L)
+    const ASN1_TIME *at = X509_CRL_get0_nextUpdate(crl);
+#else
+    ASN1_TIME *at = crl->crl->nextUpdate;
+#endif
+
+    ASN1_GENERALIZEDTIME *gt = ASN1_TIME_to_generalizedtime(at, NULL);
+    if (!gt)
+    {
+        LogError("crl could not find field\n");
+        return 0;
+    }
+    // LogInfo("crl data %s\n", (char*)gt->data);
+
+    // "20181011181119Z"
+    int success = gt->length >= 14 ? 1 : 0;
+    struct tm tm = { 0, };
+
+    if (success)
+    {
+        tm.tm_year = atoin((char*)gt->data, 0, 4) - 1900;
+        tm.tm_mon = atoin((char*)gt->data, 4, 2) - 1;
+        tm.tm_mday = atoin((char*)gt->data, 6, 2);
+        tm.tm_hour = atoin((char*)gt->data, 8, 2);
+        tm.tm_min = atoin((char*)gt->data, 10, 2);
+        tm.tm_sec = atoin((char*)gt->data, 12, 2);
+
+        success = (tm.tm_year > 100) &&
+            (tm.tm_mon >= 0 && tm.tm_mon  < 12) &&
+            (tm.tm_mday > 0 && tm.tm_mday < 32) &&
```

Comment:
you are sure that days differently from months and other times start with 1, right?:) weird :)

Reply:
yes, this is weird. (I double checked the docs.)